backend/internal/group/store/store.go

```diff
@@ -0,0 +1,507 @@
+	if err != nil {
+		return model.Group{}, err
+	}
+	group.Groups = *childGroups
+
+	// Get users
+	users, err := getGroupUsers(dbClient, id, logger)
+	if err != nil {
+		return model.Group{}, err
+	}
+	group.Users = users
+
+	return group, nil
+}
+
+// UpdateGroup updates an existing group.
+func UpdateGroup(group model.Group) error {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "GroupStore"))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("identity")
+	if err != nil {
+		logger.Error("Failed to get database client", log.Error(err))
+		return fmt.Errorf("failed to get database client: %w", err)
+	}
+	defer func() {
+		if closeErr := dbClient.Close(); closeErr != nil {
+			logger.Error("Failed to close database client", log.Error(closeErr))
+		}
+	}()
+
+	// Check for name conflicts (excluding current group)
+	err = checkGroupNameConflict(dbClient, group.Name, group.Parent, group.ID, logger)
+	if err != nil {
+		return err
+	}
+
+	// Determine the parent group ID and OU ID
+	var parentGroupID *string
+	var ouID string
+
+	if group.Parent.Type == GroupType {
+		parentGroupID = &group.Parent.ID
+		// Get the OU id from the parent group
+		parentGroup, err := GetGroup(group.Parent.ID)
+		if err != nil {
+			logger.Error("Failed to get parent group", log.Error(err))
+			return model.ErrParentNotFound
+		}
+		// Convert Group to GroupBasic for getOUFromPath function
+		parentGroupBasic := model.GroupBasic{
+			ID:          parentGroup.ID,
+			Name:        parentGroup.Name,
+			Description: parentGroup.Description,
+			Parent:      parentGroup.Parent,
+		}
+		ouID = getOUFromPath(parentGroupBasic)
+	} else {
+		ouID = group.Parent.ID
+	}
+
+	// Generate path
+	path := generateGroupPath(group.Name, group.Parent)
+
+	rowsAffected, err := dbClient.Execute(
+		QueryUpdateGroup,
+		group.ID,
+		parentGroupID,
+		ouID,
+		group.Name,
+		group.Description,
+		path,
+	)
+	if err != nil {
+		logger.Error("Failed to execute update group query", log.Error(err))
+		return fmt.Errorf("failed to execute query: %w", err)
+	}
+
+	if rowsAffected == 0 {
+		logger.Error("Group not found with id: " + group.ID)
+		return model.ErrGroupNotFound
+	}
+
+	// Update group users
+	err = updateGroupUsers(dbClient, group.ID, group.Users, logger)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteGroup deletes a group.
+func DeleteGroup(id string) error {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "GroupStore"))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("identity")
+	if err != nil {
+		logger.Error("Failed to get database client", log.Error(err))
+		return fmt.Errorf("failed to get database client: %w", err)
+	}
+	defer func() {
+		if closeErr := dbClient.Close(); closeErr != nil {
+			logger.Error("Failed to close database client", log.Error(closeErr))
+		}
+	}()
+
+	// Delete group users first
+	_, err = dbClient.Execute(QueryDeleteGroupUsers, id)
```

Comment:
These operations needs to be handled as transactions with rollback on error right?